site/app/repositories/forum/ThreadRepository.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+declare(strict_types=1);
+
+namespace app\repositories\forum;
+
+use app\entities\forum\Thread;
+use Doctrine\ORM\EntityRepository;
+
+class ThreadRepository extends EntityRepository {
+    /**
+     * @param int[] $category_ids
+     * @param int[] $status
+     * @return Thread[]
+     */
+    public function getAllThreads(array $category_ids, array $status, bool $get_deleted, bool $get_merged_threads, bool $filter_unread, string $user_id, int $block_number): array {
+        $block_size = 30;
+        $qb = $this->_em->createQueryBuilder();
+        $qb->select('t')
+            ->from(Thread::class, 't')
+        // AddSelect allows us to join server-side, reducing the number of database queries.
+            ->addSelect('c')
+            ->addSelect('v')
+            ->addSelect('p')
+            ->addSelect('ph')
+            ->addSelect('sf')
+            ->addSelect('u')
+            ->join('t.categories', 'c')
+            ->leftJoin('t.viewers', 'v')
+            ->leftJoin('t.posts', 'p')
+            ->leftJoin('p.history', 'ph')
+            ->leftJoin('t.favorers', 'sf')
+            ->join('t.author', 'u');
```

Comment:
Can you please use longer abbreviations for these?  Part of the reason why a lot of our existing SQL is "scary" is that the abbreviations are not clear.